kratos/utilities/math_utils.h

```diff
@@ -1215,43 +1215,46 @@ class MathUtils
      * \f$ [ s11, s22, s12 ] \f$ for 2D case.
      * @param rStressVector the given stress vector
      * @return the corresponding stress tensor in matrix form
+     * @tparam TVector The vector type considered
+     * @tparam TMatrixType The matrix returning type
      */
-    static inline MatrixType StressVectorToTensor(const Vector& rStressVector)
+   template<class TVector, class TMatrixType = MatrixType>
+    static inline TMatrixType StressVectorToTensor(const TVector& rStressVector)
     {
-        KRATOS_TRY
-        Matrix StressTensor;
+        KRATOS_TRY;
+        TMatrixType stress_tensor;
 
         if (rStressVector.size()==3) {
-            StressTensor.resize(2,2,false);
-            StressTensor(0,0) = rStressVector[0];
-            StressTensor(0,1) = rStressVector[2];
-            StressTensor(1,0) = rStressVector[2];
-            StressTensor(1,1) = rStressVector[1];
+            stress_tensor.resize(2,2,false);
+            stress_tensor(0,0) = rStressVector(0);
```

Comment:
Why the change from `[]` to `()`? Shouldn't be the same or has something to do with the AMatrix PR?

Reply:
This is the way to be compatible with AMatrix and to work always with all the arrays/vectors. Trust me I tried